src/lib/objects/split/blockchain.ts

```diff
@@ -1,17 +1,35 @@
-import { Interface } from 'ethers'
-import type { SplitterFactoryContract, SplitterContract, GetContract } from './types'
-import splitterFactoryAbi from './abis/splitter-factory.json'
-import splitterAbi from './abis/splitter.json'
+import { Interface, type Provider } from 'ethers'
+import type { GetContract } from './types'
+import splitterFactoryAbi from './contracts/abis/splitter-factory.json'
+import splitterAbi from './contracts/abis/splitter.json'
 import { defaultBlockchainNetwork } from '$lib/adapters/transaction'
 import type { Balance } from './schemas'
+import type { Splitter, SplitterFactory } from './contracts/types'
 
 const splitterFactoryAddress = defaultBlockchainNetwork.objects.splitterFactory
```

Comment:
I just realized that the `splitterFactory` address is hardcoded in the `BlockchainNetwork` type. I don't think that's a good place to store it, it seems it would be better if the object code would define it, especially if the object is an external type then there is no way that it can expect the host application to know about it and expose it in the SDK.

This does not belong in this PR, but I just wanted to document this issue somewhere.

Reply:
I kinda agree. At the same time, it also depends on which blockchain the app is connected to. I'll create an issue about this but I believe there is some discussion needed.